src/iks/iks.py

```diff
@@ -110,20 +126,40 @@ def iks(args):
     else:
         # handle the case of using python
         # 1. Domain Name and DNS
+        
         user_DNS = DNSCreator(UserInfo.crn, UserInfo.zone_id, UserInfo.api_endpoint, UserInfo.app_url)
         user_DNS.create_records()
+        
         # 2. Generate certificate in manager if necessary
+        UserInfo.cert_name="cis-cert"
+        
         cms_id = UserInfo.get_cms()
         print("\n"+cms_id)
         user_cert = SecretCertificateCreator(
             cis_crn=UserInfo.crn, 
             cluster_id=UserInfo.iks_cluster_id, 
             cis_domain=UserInfo.cis_domain, 
             cert_manager_crn=cms_id,
-            token=UserInfo.token["access_token"]
+            token=UserInfo.token["access_token"],
+            cert_name=UserInfo.cert_name
             )
         user_cert.create_secret()
 
+       
+        
+        #3 generate ingress
+        UserInfo.secret_name=UserInfo.cert_name
+        user_ingress = IngressCreator(
+            clusterNameOrID=UserInfo.iks_cluster_id,
+            resourceGroupID=UserInfo.get_resource_id(), 
```

Comment:
At this point in the execution, get_resource_id should have been called already in handle_args, so you could just use UserInfo.resource_id. You don't need to request the id again